js/piwik.js

```diff
@@ -3223,7 +3225,14 @@ if (typeof window.Piwik !== 'object') {
                     (msToExpire ? ';expires=' + expiryDate.toGMTString() : '') +
                     ';path=' + (path || '/') +
                     (domain ? ';domain=' + domain : '') +
-                    (secure ? ';secure' : '');
+                    (isSecureCookie ? ';secure' : '');
+            }
+
+            /*
+             * Set cookie secure flag
+             */
+            function setSecureCookie(value) {
+                isSecureCookie = value == 1 ? true : false;
```

Comment:
Can see them on the PR as well, but could you try to rebase your branch and rebuilt the minified file? It's currently conflicting with 3.x-dev branch.